test/integration/404-page/test/index.test.js

```diff
@@ -108,6 +108,44 @@ describe('404 Page Support', () => {
     runTests('serverless')
   })
 
+  it('should not cache for custom 404 page with ssg', async () => {
+    await fs.move(pages404, `${pages404}.bak`)
+    await fs.writeFile(
+      pages404,
+      `
+      const page = () => 'custom 404 page'
+      export async function getStaticProps() { return { props: {} } }
```

Comment:
It would probably be good to add a test case for revalidate as well since that's allowed currently